server/sonar-server/src/test/java/org/sonar/server/ce/ws/TaskFormatterTest.java

```diff
@@ -185,6 +187,47 @@ public void formatActivities() {
     assertThat(wsTasks).extracting("id").containsExactly("UUID1", "UUID2");
   }
 
+  @Test
+  public void formatActivity_with_both_error_message_and_stacktrace() {
+    CeActivityDto dto = newActivity("UUID", "COMPONENT_UUID", CeActivityDto.Status.FAILED)
+        .setErrorMessage("error msg")
+        .setErrorStacktrace("error stacktrace");
+    setHasErrorStacktrace(dto, true);
+
+    WsCe.Task task = underTest.formatActivity(db.getSession(), Collections.singletonList(dto)).iterator().next();
+
+    assertThat(task.getErrorMessage()).isEqualTo(dto.getErrorMessage());
+    assertThat(task.getErrorStacktrace()).isEqualTo(dto.getErrorStacktrace());
+    assertThat(task.getHasErrorStacktrace()).isTrue();
+  }
+
+  @Test
+  public void formatActivity_with_both_error_message_only() {
+    CeActivityDto dto = newActivity("UUID", "COMPONENT_UUID", CeActivityDto.Status.FAILED)
+        .setErrorMessage("error msg");
+
+    WsCe.Task task = underTest.formatActivity(db.getSession(), Collections.singletonList(dto)).iterator().next();
+
+    assertThat(task.getErrorMessage()).isEqualTo(dto.getErrorMessage());
+    assertThat(task.hasErrorStacktrace()).isFalse();
+    assertThat(task.hasHasErrorStacktrace()).isTrue();
+    assertThat(task.getHasErrorStacktrace()).isFalse();
+  }
+
+  @Test
+  public void formatActivity_with_both_error_message_and_only_stacktrace_flag() {
+    CeActivityDto dto = newActivity("UUID", "COMPONENT_UUID", CeActivityDto.Status.FAILED)
+        .setErrorMessage("error msg");
+    setHasErrorStacktrace(dto, true);
+
+    WsCe.Task task = underTest.formatActivity(db.getSession(), Collections.singletonList(dto)).iterator().next();
+
+    assertThat(task.getErrorMessage()).isEqualTo(dto.getErrorMessage());
+    assertThat(task.hasErrorStacktrace()).isFalse();
+    assertThat(task.hasHasErrorStacktrace()).isTrue();
+    assertThat(task.getHasErrorStacktrace()).isTrue();
```

Comment:
Indeed, sorry, I thought that internally it was a shortcut to hasHasErrorStacktrace().
So forget it !